src/shared/components/ncUserAndGroupPicker/NcUserAndGroupPicker.vue

```diff
@@ -0,0 +1,186 @@
+import formatting from '../../../shared/mixins/formatting.js'
+import ShareTypes from '../../mixins/shareTypesMixin.js'
+
+export default {
+
+	components: {
+		NcSelect,
+	},
+
+	mixins: [ShareTypes, formatting],
+
+	props: {
+		newOwnerUserId: {
+			type: String,
+			default: '',
+		},
+		selectUsers: {
+			type: Boolean,
+			default: true,
+		},
+		selectGroups: {
+			type: Boolean,
+			default: false,
+		},
+	},
+
+	data() {
+		return {
+			query: '',
+			value: '',
+			loading: false,
+			minSearchStringLength: 1,
+			maxAutocompleteResults: 20,
+			suggestions: [],
+		}
+	},
+
+	computed: {
+		localValue: {
+			get() {
+				return this.newOwnerUserId
+			},
+			set(v) {
+				console.info('newOwnerUserId set to ', v)
+				this.$emit('update:newOwnerUserId', v)
+			},
+		},
+
+		isValidQuery() {
+			return this.query && this.query.trim() !== '' && this.query.length > this.minSearchStringLength
+		},
+
+		options() {
+			if (this.isValidQuery) {
+				return this.suggestions
+			}
+			return []
+		},
+
+		noResultText() {
+			if (this.loading) {
+				return t('tables', 'Searching …')
+			}
+			return t('tables', 'No elements found.')
+		},
+
+		userId() {
+			return getCurrentUser().uid
+		},
+	},
+
+	methods: {
+		addTransfer(selectedItem) {
+			if (selectedItem) {
+				this.localValue = selectedItem.user
+			} else {
+				this.localValue = ''
+			}
+		},
+
+		getShareTypes() {
+			if (this.selectUsers && this.selectGroups) {
+				return [
+					this.SHARE_TYPES.SHARE_TYPE_USER,
+					this.SHARE_TYPES.SHARE_TYPE_GROUP,
+				]
+			} else if (this.selectUsers) {
+				return [
+					this.SHARE_TYPES.SHARE_TYPE_USER,
+				]
+			} else if (this.selectGroups) {
+				return [
+					this.SHARE_TYPES.SHARE_TYPE_GROUP,
+				]
+			} else {
+				return []
+			}
+		},
+
+		async asyncFind(query) {
+			this.query = query.trim()
+			if (this.isValidQuery) {
+				this.loading = true
+				await this.debounceGetSuggestions(query)
+			}
+		},
+
+		async getSuggestions(search) {
+			this.loading = true
+			const shareTypes = this.getShareTypes()
+			let shareTypeQueryString = ''
+			shareTypes.forEach(shareType => {
+				shareTypeQueryString += `&shareTypes[]=${shareType}`
+			})
+			const url = generateOcsUrl('core/autocomplete/get?search={searchQuery}&itemType=%20&itemId=%20{shareTypeQueryString}&limit={limit}', { searchQuery: search, shareTypeQueryString, limit: this.maxAutocompleteResults })
+			await axios.get(url).then(res => {
+				const rawSuggestions = res.data.ocs.data.map(autocompleteResult => {
+					return {
+						user: autocompleteResult.id,
+						displayName: autocompleteResult.label,
+						icon: autocompleteResult.icon,
+						isUser: autocompleteResult.source.startsWith('users'),
+						key: autocompleteResult.source + '-' + autocompleteResult.id,
+					}
+				})
+
+				this.suggestions = this.filterOutCurrentUser(rawSuggestions)
+
+				this.loading = false
+
+			}).catch(err => {
+				console.debug(err)
+			})
```

Comment:
Await is not used currently, might be better to wrap with try/catch and actually await the axios result
```suggestion
			try {
				const res = await axios.get(url)
				const rawSuggestions = res.data.ocs.data.map(autocompleteResult => {
					return {
						user: autocompleteResult.id,
						displayName: autocompleteResult.label,
						icon: autocompleteResult.icon,
						isUser: autocompleteResult.source.startsWith('users'),
						key: autocompleteResult.source + '-' + autocompleteResult.id,
					}
				})

				this.suggestions = this.filterOutCurrentUser(rawSuggestions)

				this.loading = fals
			} catch (err) {
				console.debug(err)
				// TODO: Should at least show some error to the user that the request failed
				// e.g. with https://github.com/nextcloud-libraries/nextcloud-dialogs#toasts
				showError(t('tables', 'Failed to fetch users'))
			}
```